### PR TITLE
SNOW-1842839: Add support for Series.str.pad

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1842839

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [ ] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://github.com/snowflakedb/snowpark-python/blob/main/CONTRIBUTING.md#thread-safe-development)

3. Please describe how your code solves the related issue.

   Add support for Series.str.pad.

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 1]
```diff
@@ -16443,8 +16443,15 @@ def str_pad(
...
-        ErrorMessage.method_not_implemented_error("pad", "Series.str")
+    ) -> "SnowflakeQueryCompiler":
+        if side == "left":
+            return self.str_rjust(width, fillchar)
+        elif side == "right":
+            return self.str_ljust(width, fillchar)
+        elif side == "both":
+            return self.str_center(width, fillchar)
+        else:
+            raise ValueError("Invalid side")
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:16454]: Nit: Can we have this ValueError mention the API?
```suggestion
            raise ValueError("Invalid side for Series.str.pad")
```

**Review by sfc-gh-nkrishna — APPROVED** [review]
One nit, but LGTM! Nice to see how it all ties together :)

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:16454]: This is mimicking the exact error message raised by native pandas! so I didn't want to change it.

**Review by sfc-gh-joshi — APPROVED** [review]

**Review by sfc-gh-lmukhopadhyay — APPROVED** [review]
LGTM!

**Pull request merged**